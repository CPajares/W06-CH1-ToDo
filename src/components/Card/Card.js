import "./Card.css";

const Card = ({ id, name, state, onclick }) => {
  return (
    <>
      <h2>Task:</h2>
      <li className="task">
        <p>{name}</p>
        <p>{state ? "pending" : "finished"}</p>
        <button className="btn btn-primary" type="button">
          update
        </button>
        <button onClick={onclick} className="btn btn-danger" type="button">
          delete
        </button>
      </li>
    </>
  );
};

export default Card;
