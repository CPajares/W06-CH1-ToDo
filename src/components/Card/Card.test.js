import Card from "./Card";
import ReactTestRenderer from "react-test-renderer";

describe("Given the Card component", () => {
  describe("When itÂ´s called with the name patata", () => {
    test("Should render a Card with name patata", () => {
      const card = ReactTestRenderer.create(
        <Card name={"patata"} state={true} />
      );
      expect(card.toJSON()).toMatchSnapshot();
    });
  });
});
