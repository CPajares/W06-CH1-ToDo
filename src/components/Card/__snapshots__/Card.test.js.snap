// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Card component When itÂ´s called with the name patata Should render a Card with name patata 1`] = `
Array [
  <h2>
    Task:
  </h2>,
  <li
    className="task"
  >
    <p>
      patata
    </p>
    <p>
      pending
    </p>
    <button
      type="button"
    >
      update
    </button>
    <button
      type="button"
    >
      delete
    </button>
  </li>,
]
`;

exports[`dsfdsf sdgdsgdfsg dfgdfg 1`] = `
Array [
  <h2>
    Task:
  </h2>,
  <li
    className="task"
  >
    <p>
      patata
    </p>
    <p>
      pending
    </p>
    <button
      type="button"
    >
      update
    </button>
    <button
      type="button"
    >
      delete
    </button>
  </li>,
]
`;
